response_operations_ui/views/messages.py

```diff
@@ -206,29 +242,39 @@ def view_selected_survey(selected_survey):
         limit = request.args.get('limit', default=10, type=int)
         flash_message = request.args.get('flash_message', default="", type=str)
         conversation_tab = request.args.get('conversation_tab', default='open')
-
-        thread_count = message_controllers.get_conversation_count(
-            {'survey': survey_id,
-             'is_closed': conversation_tab == 'closed',
-             'my_conversations': conversation_tab == 'my messages',
-             'new_respondent_conversations': conversation_tab == 'initial'})
+        # ru_ref = request.args.get('ru_ref', default='')
+        ru_ref_filter = request.args.get('ru_ref_filter', default='')
+        # business_id = request.args.get('business_id', default='')
+        business_id_filter = request.args.get('business_id_filter', default='')
+        
+        form = SecureMessageRuFilterForm()
+
+        if form.validate_on_submit():
+            new_ru_ref = form.ru_ref_filter.data
+            if new_ru_ref and new_ru_ref != ru_ref_filter:
+                business_id_filter = party_controller.try_get_party_id_by_ru_ref(new_ru_ref)
```

Comment:
Here you should use the original `get_party_by_ru_ref` function that is tested and does error handling for you.
For this I know you don't want it go to bang if it's not found, you can catch the error and handle it appropriately.  Somthing like
```
try:
    business_id_filter = party_controller.get_party_id_by_ru_ref(new_ru_ref)
except ApiError as e:
    if . e.status_code == 404:
        pass #I think pass lets you just carry on and not care?
    else:
         raise
```

That might need a bit of tweaking to be more terse, but you can just handle the error, check if it's a 404, and not care if it is.

Reply:
Nope, disagree , see reasoning in comment above

Reply:
Even after a while away I still think having 2 functions that do basically the same thing is wrong.  I updated the example to be cleaner.  If you still disagree then a 3rd party might be better to decide.  If they think my way isn't right, then your new function needs to handle 400/5XX errors as it currently doesn't do that.
```
try:
    business_id_filter = party_controller.get_party_id_by_ru_ref(new_ru_ref)
except ApiError as e:
    if e.status_code != 404:
        raise
    business_id_filter = ""
```

Reply:
I have to agree with Adam here, we shouldnt be creating an identical method without error handling. Devs will end up using it, if accidentally, and that will cause us more problems later on.

Reply:
Its been moved to be a ui functin that calls the existing endpoint and handles any exceptions it throws